test/e2e/operator_test.go

```diff
@@ -228,3 +249,199 @@ func setImage(deployment *appsv1.Deployment, image string) {
+	}
+	var upstreamIP string
+	if len(upstreamSvc.Spec.ClusterIP) != 0 {
+		upstreamIP = upstreamSvc.Spec.ClusterIP
+	} else {
+		t.Fatalf("failed to get clusterIP for service %s/%s", upstreamSvc.Namespace, upstreamSvc.Name)
+	}
+
+	// Update cluster DNS forwarding with the upstream resolver's Service IP address.
+	defaultDNS := &operatorv1.DNS{}
+	if err := cl.Get(context.TODO(), types.NamespacedName{Name: operatorcontroller.DefaultDNSController}, defaultDNS); err != nil {
+		t.Fatalf("failed to get default dns: %v", err)
+	}
+	upstream := operatorv1.Server{
+		Name:  "test",
+		Zones: []string{"foo.com"},
+		ForwardPlugin: operatorv1.ForwardPlugin{
+			Upstreams: []string{upstreamIP},
+		},
+	}
+	defaultDNS.Spec.Servers = []operatorv1.Server{upstream}
+	if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+		t.Fatalf("failed to update dns %s: %v", defaultDNS.Name, err)
+	}
+	defer func() {
+		defaultDNS = &operatorv1.DNS{}
+		if err := cl.Get(context.TODO(), types.NamespacedName{Name: "default"}, defaultDNS); err != nil {
+			t.Fatalf("failed to get default dns: %v", err)
+		}
+		if len(defaultDNS.Spec.Servers) != 0 {
+			// dnses.operator/default has a nil spec by default.
+			defaultDNS.Spec = operatorv1.DNSSpec{}
+			if err := cl.Update(context.TODO(), defaultDNS); err != nil {
+				t.Fatalf("failed to update dns %s: %v", defaultDNS.Name, err)
+			}
+		}
+	}()
+
+	// Verify that the Corefile of DNS DaemonSet pods have been updated.
+	dnsDaemonSet := &appsv1.DaemonSet{}
+	if err := cl.Get(context.TODO(), operatorcontroller.DNSDaemonSetName(defaultDNS), dnsDaemonSet); err != nil {
+		fmt.Errorf("failed to get daemonset %s/%s: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	selector, err := metav1.LabelSelectorAsSelector(dnsDaemonSet.Spec.Selector)
+	if err != nil {
+		t.Fatalf("daemonset %s/%s has invalid spec.selector: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	defaultDNSPods := &corev1.PodList{}
+	if err := cl.List(context.TODO(), defaultDNSPods, client.MatchingLabelsSelector{Selector: selector}, client.InNamespace(dnsDaemonSet.Namespace)); err != nil {
+		t.Fatalf("failed to list pods for dns daemonset %s/%s: %v", dnsDaemonSet.Namespace, dnsDaemonSet.Name, err)
+	}
+	catCmd := []string{"cat", "/etc/coredns/Corefile"}
+	for _, pod := range defaultDNSPods.Items {
+		if err := lookForStringInPodExec(pod.Namespace, pod.Name, "dns", catCmd, upstreamIP, 2*time.Minute); err != nil {
+			t.Fatalf("failed to find %s in %s of pod %s/%s: %v", upstreamIP, catCmd[1], pod.Namespace, pod.Name, err)
+		}
+	}
+
+	// Get the openshift-cli image.
+	var (
+		cliImage      string
+		cliImageFound bool
+	)
+	for _, ver := range co.Status.Versions {
+		if ver.Name == operatorcontroller.OpenshiftCLIVersionName {
+			if len(ver.Version) == 0 {
+				t.Fatalf("clusteroperator %s has an empty %s version", operatorcontroller.DNSOperatorName, operatorcontroller.OpenshiftCLIVersionName)
```

Comment:
Alternatively, you can replace the `t.Fatalf` with a `break` and fall through to the `if !cliImageFound` check below.

Reply:
ack

Reply:
ack